triton-cli/src/main.rs

```diff
@@ -89,171 +89,122 @@ fn parse_inputs(inputs: Option<String>) -> Vec<BFieldElement> {
     let stark = Stark {
-        security_level: read_usize(&mut reader),
-        fri_expansion_factor: read_usize(&mut reader),
-        num_trace_randomizers: read_usize(&mut reader),
-        num_collinearity_checks: read_usize(&mut reader),
-        num_combination_codeword_checks: read_usize(&mut reader),
+        security_level: usize::try_from(serialized_proof.security_level)?,
+        fri_expansion_factor: usize::try_from(serialized_proof.fri_expansion_factor)?,
+        num_trace_randomizers: usize::try_from(serialized_proof.num_trace_randomizers)?,
+        num_collinearity_checks: usize::try_from(serialized_proof.num_colinearity_checks)?,
+        num_combination_codeword_checks: usize::try_from(
+            serialized_proof.num_combination_codeword_checks,
+        )?,
     };
-    let digest_len = read_u8(&mut reader);
-    let input_len = read_u64(&mut reader);
-    let output_len = read_u64(&mut reader);
-    let proof_len = read_u64(&mut reader);
-    let mut d = Digest::default();
+    let mut digest = Digest::default();
     assert_eq!(
-        d.0.len(),
-        usize::from(digest_len),
+        digest.0.len(),
+        serialized_proof.program_digest.len(),
         "digest length mismatch!"
     );
-    read_u64_vec(&mut reader, digest_len.into())
+    serialized_proof
+        .program_digest
         .iter()
         .enumerate()
-        .for_each(|(i, x)| d.0[i] = BFieldElement::new(*x));
+        .for_each(|(i, x)| digest.0[i] = BFieldElement::new(*x));
     let claim = Claim {
-        program_digest: d,
-        input: read_u64_vec(&mut reader, input_len)
+        program_digest: digest,
+        input: serialized_proof
+            .public_inputs
             .iter()
             .map(|v| BFieldElement::new(*v))
             .collect(),
-        output: read_u64_vec(&mut reader, output_len)
+        output: serialized_proof
+            .public_outputs
             .iter()
             .map(|v| BFieldElement::new(*v))
             .collect(),
     };
-    let proof_vec = read_u64_vec(&mut reader, proof_len)
+    let proof_vec = serialized_proof
+        .proof
         .iter()
         .map(|v| BFieldElement::new(*v))
         .collect();
     Ok((stark, claim, Proof(proof_vec)))
 }
 
-fn write_proof(data: (Stark, Claim, Proof), out: &String) -> Result<()> {
+fn write_proof(data: (Stark, Claim, Proof), out: &str) -> Result<()> {
     let (stark, claim, proof) = data;
+    let serialized = SerializedProof {
+        version: 1,
+        security_level: u64::try_from(stark.security_level)?,
+        fri_expansion_factor: u64::try_from(stark.fri_expansion_factor)?,
+        num_trace_randomizers: u64::try_from(stark.num_trace_randomizers)?,
+        num_colinearity_checks: u64::try_from(stark.num_collinearity_checks)?,
+        num_combination_codeword_checks: u64::try_from(stark.num_combination_codeword_checks)?,
+        program_digest: claim.program_digest.0.iter().map(u64::from).collect(),
```

Comment:
Maybe (but it's not important):

```suggestion
        program_digest: claim.program_digest.values().map(u64::from).collect(),
```